app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -64,118 +60,92 @@ class TorManager @Inject constructor(
      * Starts [ShareService] and creates a new onion service.
      * Suspends until the address of the onion service is available.
      */
+    @Throws(IOException::class, IllegalArgumentException::class, InterruptedException::class)
     suspend fun start() = withContext(Dispatchers.IO) {
-        if (state.value !is TorState.Stopped) stop()
-
         LOG.info("Starting...")
         val now = System.currentTimeMillis()
         _state.value = TorState.Starting(progress = 0, startTime = now, lastProgressTime = now)
-
         Intent(app, ShareService::class.java).also { intent ->
             startForegroundService(app, intent)
         }
 
         tor.start()
+        changeStartingState(5)
+        if (settingsManager.automaticBridges.value) {
+            startCheckJob = launch {
+                LOG.info("Starting check job")
+                checkStartupProgress()
+                LOG.info("Check job finished")
+            }
+        } else {
+            val customBridges = settingsManager.customBridges.value
+            if (customBridges.isNotEmpty()) {
+                LOG.info("Using ${customBridges.size} custom bridges...")
+                tor.enableBridges(customBridges.map { "Bridge $it" })
+            }
+        }
+        tor.enableNetwork(true)
     }
 
-    fun stop(failedToConnect: Boolean = false) {
+    fun stop() {
         LOG.info("Stopping...")
         startCheckJob?.cancel()
         startCheckJob = null
-        _state.value = TorState.Stopping(failedToConnect)
```

Comment:
Is this now being done elsewhere, or do we not use the `Stopping` state any more?

Reply:
There's no more `TorState.Stopping` in the current iteration. But there is `TorState.FailedToConnect` now to handle the case where we need to communicate connection failure to the UI.